examples/RecordBuilder/README.md

```diff
@@ -0,0 +1,118 @@
+
+# Record Builder 
+
+This example demonstrates the Record Builder pattern in Roc. This pattern leverages the functional programming concept of applicative functors, to provide a flexible method for constructing complex types.
+
+## The Basics
+
+Let's assume we want to develop a module that supplies a type-safe yet versatile method for users to safely obtain sequential IDs. The record builder pattern can be beneficial in this scenario.
+
+1. **Opaque Type** We need an opaque type that will accumulate our state:
+```roc
+Count a := (U32, a)
+```
+This type takes a type variable `a`, and holds a tuple of a `U32` and an `a`. Here, U32 will maintain our counter's value, while a houses the function to progress our builder pattern.
```

Comment:
```suggestion
This type takes a type variable `a`, and holds a tuple of a `U32` and an `a`. Here, U32 will maintain our counter's value, while `a` houses the function to progress our builder pattern.
```

Reply:
Maybe we shouldn't just say `a` will house the function. It holds the state we are composing, which is a function if we haven't yet provided all the values or the final value if we have done so.